### PR TITLE
Improve READMEs

### PR DESCRIPTION
**Pull request by jean-philippe-martin**
- fix merge conflict
- fix broken link
- fill in first draft of gcloud-java-nio README

### CODE REVIEW

File: gcloud-java-contrib/gcloud-java-nio/README.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+If you know the paths will point to Google Cloud Storage, you can also use the direct
+formulation:
+
+    try (CloudStorageFileSystem fs = CloudStorageFileSystem.forBucket("bucket") {
+          Path path = fs.getPath("lolcat.csv");
+          List<String> lines = Files.readAllLines(path, StandardCharsets.UTF_8);
+    }
+
+Once you have a `Path` you can use it as you would a normal file. For example
+you can use InputStream and OutputStream for streaming:
```
Comment (mziccard) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: InputStream -> `InputStream`
OutputStream -> `InputStream`

File: gcloud-java-contrib/gcloud-java-nio/README.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+    Path path = Paths.get(URI.create("gs://bucket/lolcat.csv"));
+    List<String> lines = Files.readAllLines(path, StandardCharsets.UTF_8);
+
+If you know the paths will point to Google Cloud Storage, you can also use the direct
+formulation:
+
+    try (CloudStorageFileSystem fs = CloudStorageFileSystem.forBucket("bucket") {
+          Path path = fs.getPath("lolcat.csv");
+          List<String> lines = Files.readAllLines(path, StandardCharsets.UTF_8);
```
Comment (mziccard) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Here and in the line above, indentation should be just 2 spaces:
```
try (CloudStorageFileSystem fs = CloudStorageFileSystem.forBucket("bucket") {
  Path path = fs.getPath("lolcat.csv");
  List<String> lines = Files.readAllLines(path, StandardCharsets.UTF_8);
}
```

File: gcloud-java-contrib/gcloud-java-nio/README.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+================================================
+
+Java idiomatic NIO client for [Google Cloud Storage](https://cloud.google.com/storage/).
+
+This client library allows you to easily interact with Google Cloud Storage using Java's
+standard file system API.
+
+[![Build Status](https://travis-ci.org/GoogleCloudPlatform/gcloud-java.svg?branch=master)](https://travis-ci.org/GoogleCloudPlatform/gcloud-java)
+[![Coverage Status](https://coveralls.io/repos/GoogleCloudPlatform/gcloud-java/badge.svg?branch=master)](https://coveralls.io/r/GoogleCloudPlatform/gcloud-java?branch=master)
+[![Maven](https://img.shields.io/maven-central/v/com.google.cloud/gcloud-java-nio.svg)]( https://img.shields.io/maven-central/v/com.google.cloud/gcloud-java-nio.svg)
```
Comment (mziccard) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Could you also add Codacy and dependency status badges?
```
 [![Codacy Badge](https://api.codacy.com/project/badge/grade/9da006ad7c3a4fe1abd142e77c003917)](https://www.codacy.com/app/mziccard/gcloud-java)
 [![Dependency Status](https://www.versioneye.com/user/projects/56bd8ee72a29ed002d2b0969/badge.svg?style=flat)](https://www.versioneye.com/user/projects/56bd8ee72a29ed002d2b0969)		 [![Dependency Status](https://www.versioneye.com/user/projects/56bd8ee72a29ed002d2b0969/badge.svg?style=flat)](https://www.versioneye.com/user/projects/56bd8ee72a29ed002d2b0969)
```

File: gcloud-java-contrib/gcloud-java-nio/README.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+**NOTE:** Cloud Storage uses a flat namespace and therefore doesn't support real
+directories. So this library supports what's known as "pseudo-directories". Any path that
+includes a trailing slash, will be considered a directory. It will always be assumed to exist,
+without performing any I/O. This allows you to do path manipulation in the same manner as you
+would with the normal UNIX file system implementation. You can disable this feature with
+`CloudStorageConfiguration.usePseudoDirectories()`.
+
+
+
```
Comment (mziccard) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: You can remove a couple of empty lines above here

File: gcloud-java-contrib/gcloud-java-nio/README.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+* [`ParallelCountBytes`](../../gcloud-java-examples/src/main/java/com/google/cloud/examples/nio/ParallelCountBytes.java) efficiently downloads a file from Google Cloud Storage.
+
+* [`ListFileSystems`](../gcloud-java-nio-examples/README.md) illustrates how NIO can add Google Cloud Storage support to some legacy programs, without having to modify them.
+
+
+Authentication
+--------------
+
+See the [Authentication](https://github.com/GoogleCloudPlatform/gcloud-java#authentication) section in the base directory's README.
```
Comment (jart) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Please wrap paragraphs at 80 column.

File: gcloud-java-contrib/gcloud-java-nio/README.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+stable.
+
+Contributing
+------------
+
+Contributions to this library are always welcome and highly encouraged.
+
+See `gcloud-java`'s [CONTRIBUTING] documentation and the `gcloud-*` [shared documentation](https://github.com/GoogleCloudPlatform/gcloud-common/blob/master/contributing/readme.md#how-to-contribute-to-gcloud) for more information on how to get started.
+
+Please note that this project is released with a Contributor Code of Conduct. By participating in this project you agree to abide by its terms. See [Code of Conduct][code-of-conduct] for more information.
```
Comment (jart) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: I'd prefer if we didn't mention this.

File: gcloud-java-contrib/gcloud-java-nio/README.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
+NIO Filesystem Provider for Google Cloud Storage
+================================================
+
+Java idiomatic NIO client for [Google Cloud Storage](https://cloud.google.com/storage/).
```
Comment (jart) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Maybe call this:

```
Implementation of Java 7 java.nio.file.FileSystem for Google Cloud...

This client library allows you to use the standardized Java file system API for interacting with Google Cloud Storage.
```

**Comment by jart** [comment]
One thing worth mentioning is we might want to consider in a future pull request merging our package-info.java documentation into the README, so we're not wasting effort.

File: gcloud-java-contrib/gcloud-java-nio/README.md (hunk 6 above)
Comment (mziccard) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Why?

File: gcloud-java-contrib/gcloud-java-nio/README.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+
+This client library allows you to easily interact with Google Cloud Storage using Java's
+standard file system API.
```
Comment (mziccard) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Should we also mention somewhere what features we do not (yet) support of NIO (like file attributes if I recall correctly)?

**Comment by jean-philippe-martin** [comment]
cc @jart, @mziccard for feedback.

File: gcloud-java-contrib/gcloud-java-nio/README.md (hunk 1 above)
Comment (jean-philippe-martin) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Done.

File: gcloud-java-contrib/gcloud-java-nio/README.md (hunk 2 above)
Comment (jean-philippe-martin) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Doen.

File: gcloud-java-contrib/gcloud-java-nio/README.md (hunk 3 above)
Comment (jean-philippe-martin) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Done (I put only one Dependency badge, I think that's what you meant).

File: gcloud-java-contrib/gcloud-java-nio/README.md (hunk 4 above)
Comment (jean-philippe-martin) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Sure. That has no impact on the rendered page, though, does it?

Comment (mziccard) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Yep, no impact on rendering

File: gcloud-java-contrib/gcloud-java-nio/README.md (hunk 5 above)
Comment (jean-philippe-martin) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Done. Some links stick out.

File: gcloud-java-contrib/gcloud-java-nio/README.md (hunk 7 above)
Comment (jean-philippe-martin) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Done.

File: gcloud-java-contrib/gcloud-java-nio/README.md (hunk 3 above)
Comment (mziccard) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Why not the `Codacy` badge?

File: gcloud-java-contrib/gcloud-java-nio/README.md (hunk 8 above)
Comment (jean-philippe-martin) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Yes, absolutely. I've added a little something, but we'll probably want to refine it.

**Comment by jean-philippe-martin** [comment]
@jart, agreed on package-info.java. Ideally we don't repeat ourselves, but have a mechanism to pull text from one onto the other? Alternatively, the package-info.java could just be a link to the readme, though of course that's not as good.

**Comment by jean-philippe-martin** [comment]
I've also added "(alpha)" to the title, to temper expectations.

File: gcloud-java-contrib/gcloud-java-nio/README.md (hunk 3 above)
Comment (jean-philippe-martin) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: The snipped you gave me had one Codacy and two Dependency badges. I put the Codacy badge, but only one Dependency badge.

Comment (mziccard) [inline gcloud-java-contrib/gcloud-java-nio/README.md]: Yes, sorry, I just noticed. My bad!

**Pull request merged**